shell/gpu/gpu_surface_metal.h

```diff
@@ -17,7 +17,11 @@
 
 namespace flutter {
 
+#if TARGET_IPHONE_SIMULATOR
+class API_AVAILABLE(ios(13.0)) GPUSurfaceMetal : public Surface {
```

Comment:
Can you use `SK_API_AVAILABLE_CA_METAL_LAYER` here please? There is currently an issue with the compiler version we use where decorating C++ classes with API_AVAILABLE marks the dylib symbol visibility as default instead of hidden. This is why we added the SK_API_AVAILABLE macro is Skia as well. That macro is wrapped by SK_API_AVAILABLE_CA_METAL_LAYER. Just using that macro directly should give you the benefit of not making dylib symobls visible and also the if-else macros.